src/tiled/objectselectiontool.cpp

```diff
@@ -551,8 +552,22 @@ void ObjectSelectionTool::mouseReleased(QGraphicsSceneMouseEvent *event)
             break;
         }
         const Qt::KeyboardModifiers modifiers = event->modifiers();
+        QSet<MapObjectItem*> selection = mapScene()->selectedObjectItems();
+        if (modifiers & Qt::AltModifier) {
+            auto underlyingObjects = listOfObjectItemsAt(event->scenePos());
+
+            if (!underlyingObjects.size())
+                break;
+            if (mSelectedRotationIndex >= underlyingObjects.size())
+                mSelectedRotationIndex = 0;//definitely another stack
+            selection.clear();
+            selection.insert(underlyingObjects.at(mSelectedRotationIndex));
```

Comment:
Note that Tiled adds to the selection when Shift or Control modifiers are held. The same should apply here, so you can use Shift or Control to add below objects to the selection.

I have a hunch, btw, that you can re-use the code below by just changing the `mClickedObjectItem` to the next underlying object.